### PR TITLE
Decorate Class

### PR DESCRIPTION
**Pull request by naanmohammed**
For this millestone I :

-   Created a class Nameable with a method correct_name that does the following:
-   Added a method validate_name that uses the Corrector instance to validate and save the corrected name.

### CODE REVIEW

**Review by DammyShittu — COMMENTED** [review]
Hi @naanmohammed,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### Your readme documentation should follow the template and should contain all the main headers on it.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by zainsadaqat — COMMENTED** [review]
# ❌ ❌ ❌Invalid Code Review Request ❌ ❌ ❌
# Hello @naanmohammed  👋  

## Something went wrong 

### While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

Your Code Review Request is marked as invalid for this reason:

- ### You have submitted a project, where you've not attached the required linters. Please attach the linters and submit for a review again. 👍 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by rloterh — COMMENTED** [review]
Hi Naan, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, whereby required linter checks are not showing on the Pull Request. Kindly ensure to resolve branch conflicts and that GitHub actions are properly configured with regard to your project and that **all** required linters are showing green.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require_relative 'base_decorator'
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    super.capitalize
```
Comment (Deepakdanger) [inline capitalize_decorator.rb]: - [ ] Instead of `super.capitalize` kindly use `@nameable.correct_name.capitalize`

File: base_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class BaseDecorator < Nameable
```
Comment (Deepakdanger) [inline base_decorator.rb:1]: - [ ] Kindly include `attr_accessor :nameable` here.

**Comment by naanmohammed** [comment]
Please note the latest commit [Merge branch 'dev' into decorate](https://github.com/naanmohammed/oop_school_library/pull/2/commits/b75d89705d0f43d3ddbe19486ffb8398e7cd7356) was made to resolve merge errors.

File: person.rb [hunk 3]
```diff
@@ -1,14 +1,24 @@
...
     @parent_permission = parent_permission
   end
 
+  def correct_name
+    @name
+  end
+
+  def age=(age)
+    @age = age.to_i
+  end
```
Comment (Deepakdanger) [inline person.rb]: - [ ] There is no need for this method to define `@age`, we can directly use `age` anywhere in this file.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @naanmohammed , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OyePriscilla — APPROVED** [review]
Hi @naanmohammed 👍 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
Implemented a method `correct_name` 👍 
Implemented getter and setter method for basedecorator class 👍 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**